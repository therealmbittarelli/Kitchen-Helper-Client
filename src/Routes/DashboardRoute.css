.pantrytag {
  margin-top: 38%;
}

.recipetag {
  margin-top: 38%;
}

.mealplantag {
  margin-top: 38%;
}

.dashboard {
  /* display: inline-block; */
  list-style: none;
  height: 150px;
  width: 150px;
  border: 1px solid black;
  margin: 10px;
  text-align: center;
  font-size: 30px;
  vertical-align: middle; 
  background-color: #FFCB32;
  font-family: 'cookiesregular';
  box-shadow: 5px 5px 5px #2F7604;
  border-radius: 10%;
}

.dashboard:hover {
  background-color: #2F7604;
  color: #FFCB32;
}

Link {
  margin-top: 40px;
}

.dashboard-link-text {
  text-decoration: none;
  /* line-height: 140px; */
  padding: 3px; 
  color: black;
  font-weight: bold;
}
  

.dashboard.meal-plan {
  font-size: 25px;
}

#dashboard-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  vertical-align: middle;
}

#dashboard-links-list {
  text-align: center;
  justify-content: center;
  align-items: center;
  padding-right: 12.5%;
  /* margin: auto; */
}

@media(min-width: 375px) {
  #dashboard-links-list {
    text-align: center;
    justify-content: center;
    align-items: center;
    padding-right: 10.8%;
    /* margin: auto; */
  }
}

@media(min-width: 425px) {
  #dashboard-links-list {
    text-align: center;
    justify-content: center;
    align-items: center;
    padding-right: 9.5%;
    /* margin: auto; */
  }
}

@media screen and (min-width: 700px){

  .dashboard {
    display: inline-block;
    height: 250px;
    min-width: 250px;
  }

  .dashboard-link-text {
    /* line-height: 230px; */
    /* line-height: 185px; */
    padding: 0px;
  }

  .dashboard.meal-plan {
    font-size: 30px;
  }

  #dashboard-links-list {
    text-align: center;
    justify-content: center;
    align-items: center;
    padding-right: 5.2%;
    /* margin: auto; */
  }

  #dashboard-container {
    text-align: center;
    display: flex;
    flex-direction: row;
    justify-content: center;
  }
}

@media(min-width: 768px) {
  #dashboard-links-list {
    text-align: center;
    justify-content: center;
    align-items: center;
    padding-right: 5.2%;
    /* margin: auto; */
  }
}

@media(min-width: 1023px) {
  #dashboard-links-list {
    text-align: center;
    justify-content: center;
    align-items: center;
    padding-right: 3.9%;
    /* margin: auto; */
  }
}

@media(min-width: 1044px) {
  #dashboard-links-list {
    text-align: center;
    justify-content: center;
    align-items: center;
    padding-right: 2.8%;
    /* margin: auto; */
  }
}

@media(min-width: 1920px) {
  #dashboard-links-list {
    text-align: center;
    justify-content: center;
    align-items: center;
    padding-right: 2.1%;
    /* margin: auto; */
  }
}