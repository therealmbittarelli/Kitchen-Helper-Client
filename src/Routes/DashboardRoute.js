import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import './DashboardRoute.css';


class DashboardRoute extends Component {
    constructor(props) {
        super(props);
        this.state = {

        };
    }

    render() {
        return (
            <div id="dashboard-container">
                <ul id="dashboard-links-list">
                    <Link
                        className="dashboard-link-text"
                        to={"/pantry"}>
                        <li className="dashboard pantry">
                            <p className="pantrytag">
                                Pantry
                            </p>
                        </li>
                    </Link>
                    <Link
                        className="dashboard-link-text"
                        to={"/recipes"}>
                        <li className="dashboard recipes">
                            <p className="recipetag">
                                Recipes
                            </p>
                        </li>
                    </Link>
                    <Link
                        className="dashboard-link-text"
                        to={"/planner"}>
                        <li className="dashboard meal-plan">
                            <p className="mealplantag">
                                Plan a Meal
                            </p>
                        </li>
                    </Link>
                </ul>
            </div >
        );
    }
}

export default DashboardRoute;