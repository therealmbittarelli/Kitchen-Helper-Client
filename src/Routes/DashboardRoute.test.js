import React from 'react';
import ReactDOM from 'react-dom';
import DashboardRoute from './DashboardRoute';
import { BrowserRouter } from 'react-router-dom';
import renderer from "react-test-renderer";

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<BrowserRouter>
    <DashboardRoute />
  </BrowserRouter>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
  const tree = renderer.create(<BrowserRouter>
    <DashboardRoute />
  </BrowserRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});