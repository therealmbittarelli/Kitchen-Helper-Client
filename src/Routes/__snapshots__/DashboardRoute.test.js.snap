// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  id="dashboard-container"
>
  <ul
    id="dashboard-links-list"
  >
    <a
      className="dashboard-link-text"
      href="/pantry"
      onClick={[Function]}
    >
      <li
        className="dashboard pantry"
      >
        Pantry
      </li>
    </a>
    <a
      className="dashboard-link-text"
      href="/recipes"
      onClick={[Function]}
    >
      <li
        className="dashboard recipes"
      >
        Recipes
      </li>
    </a>
    <a
      className="dashboard-link-text"
      href="/planner"
      onClick={[Function]}
    >
      <li
        className="dashboard meal-plan"
      >
        Plan a Meal
      </li>
    </a>
  </ul>
</div>
`;
